### PR TITLE
Create a shared checkbox component

### LINKED ISSUE #3934: Experimental checkboxes are using the wrong shade of blue
image

This is one of the places where our design tokens weren't being used, so the colour didn't shift.

### PR DESCRIPTION
**Pull request by jaril**
Fix #3934.

image
image

### CODE REVIEW

File: src/ui/components/shared/SettingsModal/SettingsBodyItem.tsx [hunk 1]
```diff
@@ -27,7 +28,7 @@ function Checkbox<K>({ item, value, onChange }: InputProps<K, boolean>) {
     return <span className="italic text-gray-500">Coming Soon...</span>;
   }
 
-  return <input type="checkbox" id={String(key)} checked={value} onChange={toggleSetting} />;
+  return <_Checkbox id={String(key)} checked={value} onChange={toggleSetting} />;
```
Comment (jcmorrow) [inline src/ui/components/shared/SettingsModal/SettingsBodyItem.tsx:31]: Naming is tough here 😄 . What do you think about changing the local `CheckBox` to `CheckBoxSetting`?

File: src/ui/components/shared/UserSettingsModal/UserSettingsModal.tsx [hunk 2]
```diff
@@ -67,13 +68,12 @@ function Personal() {
           <div className="text-gray-500">{email}</div>
         </div>
       </div>
-      <div className="flex flex-row space-x-3 items-center" data-private>
-        <li className="flex flex-row items-center">
-          <label className="space-y-1.5 pr-36 flex-grow cursor-pointer" htmlFor="disableLogRocket">
+      <div className="flex flex-col" data-private>
```
Comment (jcmorrow) [inline src/ui/components/shared/UserSettingsModal/UserSettingsModal.tsx:71]: Changing the flex direction seems like an odd change, what's up here?

File: src/ui/components/shared/SettingsModal/SettingsBodyItem.tsx (hunk 1 above)
Comment (jaril) [reply src/ui/components/shared/SettingsModal/SettingsBodyItem.tsx:31]: Yep let's do that!

File: src/ui/components/shared/UserSettingsModal/UserSettingsModal.tsx (hunk 2 above)
Comment (jaril) [reply src/ui/components/shared/UserSettingsModal/UserSettingsModal.tsx:71]: Wait nevermind, there was a reason. Without `flex-col`, the row doesn't occupy the entire width:

image

**Pull request merged**